diesel/src/expression/functions/mod.rs

```diff
@@ -114,18 +119,66 @@ macro_rules! sql_function {
         sql_function!($fn_name, $struct_name, $args -> $return_type, $docs, "");
     };
 
-    ($fn_name:ident, $struct_name:ident, ($($arg_name:ident: $arg_type:ty),*)) => {
-        sql_function!($fn_name, $struct_name, ($($arg_name: $arg_type),*) -> ());
-    };
-
     (
         $fn_name:ident,
         $struct_name:ident,
         $args:tt -> $return_type:ty,
         $docs:expr,
         $helper_ty_docs:expr
     ) => {
-        sql_function_body!($fn_name, $struct_name, $args -> $return_type, $docs, $helper_ty_docs);
+        sql_function_body!($fn_name, $struct_name, $fn_name, $args -> $return_type, $docs, $helper_ty_docs);
+    };
+}
+
+#[macro_export]
+#[cfg(feature = "postgres")]
```

Comment:
Is this PG specific? `COALESCE` can have many signatures and is ANSI SQL.

Reply:
MySQL doesn't support function overloading. SQLite is an odd case. It doesn't have `CREATE FUNCTION` and functions are created through the C API. I Thought that it didn't support overloading, but after rereading the [functions page](https://www.sqlite.org/c3ref/create_function.html) it appears that they can be. I'll add a commit so that this works for SQLite.